src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExtensionPass.php

```diff
@@ -72,15 +76,18 @@ public function process(ContainerBuilder $container)
             $container->getDefinition('twig.extension.assets')->addTag('twig.extension');
         }
 
-        if (class_exists('Symfony\Component\Yaml\Parser')) {
+        $container->addResource(new ClassExistenceResource(YamlParser::class));
+        if (class_exists(YamlParser::class)) {
```

Comment:
The difference is that if one uses the `twig.extension.yaml` directly, the proposed version will break.
I think the current approach is a more open thus a better practice, even if in this very case using the extension directly is so unlikely.